src/mpi/coll/allgatherv/allgatherv.c

```diff
@@ -22,6 +22,45 @@
         The smallest message size that will be used for the pipelined, large-message,
         ring algorithm in the MPI_Allgatherv implementation.
 
+    - name        : MPIR_CVAR_ALLGATHERV_ALGORITHM_INTRA
+      category    : COLLECTIVE
+      type        : string
+      default     : auto
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        Variable to select allgatherv algorithm
+        auto - Internal algorithm selection
+        brucks - Force brucks algorithm
+        recursive_doubling - Force recursive doubling algorithm
+        ring - Force ring algorithm
+
+    - name        : MPIR_CVAR_ALLGATHERV_ALGORITHM_INTER
+      category    : COLLECTIVE
+      type        : string
+      default     : auto
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        Variable to select allgatherv algorithm
+        auto - Internal algorithm selection
+
+    - name        : MPIR_CVAR_ALLGATHERV_DEVICE_COLLECTIVE
+      category    : COLLECTIVE
+      type        : boolean
+      default     : true
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        If set to true, MPI_Allgatherv will allow the device to override the
+        default, MPIR-level collective algorithms. The device still has the
+        option to call the MPIR-level algorithms manually.
+        If set to false, the device-level allgather function will not be
```

Comment:
This needs to get fixed for all collectives. The comment still says `allgather`.